CONTRIBUTING.md

```diff
@@ -82,6 +84,76 @@ is used to keep track of public API changes.
 Run `./gradlew apiDump` to update API index files after introducing new or changing old public API. Commit updated 
 API indexes together with other changes.
 
+## Infrastructure
+
+### Java version
+
+To minimize compatibility problems, [Gradle's Java toolchains](https://docs.gradle.org/current/userguide/toolchains.html)
+are [used](build-logic/src/main/kotlin/org/jetbrains/conventions/base-java.gradle.kts) to build and test the project.
+
+When run, Gradle tries to auto-detect the required JRE/JDK installation locally, but it may fail if you don't have
+that version of Java installed or if it's installed in an unusual location. Please refer to 
```

Comment:
Maybe we can apply [org.gradle.toolchains.foojay-resolver](https://github.com/gradle/foojay-toolchains#usage) to download toolchain automatically, especially for Windows users.

Reply:
Indeed.